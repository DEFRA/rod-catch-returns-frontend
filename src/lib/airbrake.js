const { formatWithOptions, inspect } = require(('node:util'))
const { Notifier } = require('@airbrake/node')

const INSPECT_OPTS = {
  depth: null,
  maxStringLength: null,
  maxArrayLength: null,
  breakLength: null,
  compact: true,
  showHidden: true,
  errorNotifications: true
}

/** @type {Notifier} */
let airbrake = null

/**
 * Resets the Airbrake notifier instance.
 */
const reset = () => {
  airbrake = null
}

/**
 * Reports a console method invocation to Airbrake.
 *
 * @param {string} method - The console method being called (e.g., "warn" or "error").
 * @param {...any} args - The arguments passed to the console method.
 * @returns {void}
 */
const reportToAirbrake = (method, ...args) => {
  const error =
    args.find((arg) => arg instanceof Error) ??
    new Error(formatWithOptions(INSPECT_OPTS, ...args))

  const request = args.find(
    (arg) => arg && typeof arg === 'object' && 'headers' in arg
  )

  // notify returns a promise, but we do not await it
  airbrake.notify({
    error,
    params: {
      consoleInvocationDetails: {
        method,
        arguments: args.map((arg) => inspect(arg, INSPECT_OPTS))
      }
    },
    environment: {
      ...(process.env.name && { name: process.env.name })
    },
    ...(request?.state && { session: request.state }),
    context: {
      ...(request?.method && {
        action: `${request.method.toUpperCase()} ${request.path}`
      }),
      ...(request?.headers?.['user-agent'] && {
        userAgent: request.headers['user-agent']
      })
    }
  })
}

/**
 * Initializes the Airbrake client and intercepts `console.warn` and `console.error`
 * to send notifications to Airbrake/Errbit.
 *
 * If the required environment variables are not set, Airbrake will not be initialized.
 *
 * @returns {boolean} `true` if the Airbrake client was initialized, `false` otherwise.
 */
const initialise = () => {
  if (
    airbrake ||
    !process.env.AIRBRAKE_PROJECT_KEY ||
    !process.env.AIRBRAKE_HOST
  ) {
    return !!airbrake
  }

  airbrake = new Notifier({
    projectId: 4,
    projectKey: process.env.AIRBRAKE_PROJECT_KEY,
    host: process.env.AIRBRAKE_HOST,
    environment: process.env.NODE_ENV,
    errorNotifications: true,
    performanceStats: false,
    remoteConfig: false
  })

  const originalError = console.error.bind(console)
  const originalWarn = console.warn.bind(console)

  console.error = (...args) => {
    reportToAirbrake('error', ...args)
    originalError(...args)
  }

  console.warn = (...args) => {
    reportToAirbrake('warn', ...args)
    originalWarn(...args)
  }

  // Ensure uncaught errors are logged
  process.on('uncaughtExceptionMonitor', originalError)

  // Override Airbrake's uncaughtException/unhandledRejection handlers to flush before exit
  const flushAndExit = async () => {
    await airbrake.flush()
    process.exit(1)
  }
  process.on('uncaughtException', flushAndExit)
  process.on('unhandledRejection', flushAndExit)

  return true
}

/**
 * Wraps a debug logging function so that log messages are also reported to Airbrake.
 *
 * @param {Function} logFunction - The debug logging function to wrap.
 * @returns {Function} A new logging function that reports to Airbrake before logging.
 */
const attachAirbrakeToDebugLogger = (logFunction) => {
  if (!airbrake) {
    return logFunction
  }

  return (...args) => {
    reportToAirbrake(logFunction.namespace, ...args)
    logFunction(...args)
  }
}

/**
 * Flushes the Airbrake buffer, ensuring that any pending notifications are sent
 * before returning.
 *
 * @returns {Promise<void>}
 */
const flush = async () => {
  if (initialise()) {
    await airbrake.flush()
    airbrake.close()
  }
}

module.exports = {
  initialise,
  attachAirbrakeToDebugLogger,
  flush,
  reset
}
