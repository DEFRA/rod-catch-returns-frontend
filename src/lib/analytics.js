const Crypto = require('crypto')
const salt = Crypto.randomBytes(16)

const staticMatcherPublic = /^(?:\/public\/.*|\/robots.txt|\/favicon.ico)/
const isStaticResource = request => staticMatcherPublic.test(request.path)

const sessionIdProducer = (request) => {
  let hash = null
  if (!isStaticResource(request)) {
    const ip = request.info.remoteAddress
    const ua = request.headers['user-agent']

    hash = Crypto.createHash('sha256')
      .update(ip + ua)
      .update(salt)
      .digest('base64')
  }
  return hash
}
module.exports = {
  sessionIdProducer,
  isStaticResource
}
