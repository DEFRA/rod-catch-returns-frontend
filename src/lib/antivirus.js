const { logger } = require('defra-logging-facade')
const NodeClam = require('clamscan')

const wait = interval => new Promise(resolve => setTimeout(resolve, interval))

async function retryAntiVirusInit (config, retries, delay) {
  try {
    return await new NodeClam(config).init()
  } catch (err) {
    if (retries === 0) {
      throw err
    }
    logger.info(`Unable to find virus scanner - retries left ${retries}`)
    await wait(delay)
    return await retryAntiVirusInit(config, --retries, delay)
  }
}

module.exports = {
  retryAntiVirusInit
}
