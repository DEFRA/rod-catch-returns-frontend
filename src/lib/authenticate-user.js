'use strict'

const { v4: uuid } = require('uuid')

const { logger } = require('defra-logging-facade')
/**
 * Sets up the cache and the session cookies for an authenticated user
 * @param request
 * @param h
 * @returns {Promise<*>}
 */
module.exports = async (request) => {
  if (!request.app.authorization) {
    throw new Error('Cannot set up session cookie and cache for an unauthenticated user')
  }

  // Set the cookie to the new session identifier
  request.cookieAuth.set({ sid: uuid() })

  if (request.app.authorization.token) {
    // If it is a user authentication then set the authorization details in the cache
    const cache = { authorization: request.app.authorization }
    await request.cache().set(cache)
    // authorization will have the token and user's name
    logger.debug('User is authenticated: ' + JSON.stringify(request.app.authorization))
  } else {
    // If the user is authenticated by the license set the contactId
    const cache = { contactId: request.app.authorization.contactId }
    await request.cache().set(cache)
    logger.debug('Contact is authenticated: ' + JSON.stringify(request.app.authorization.contactId))
  }
}
