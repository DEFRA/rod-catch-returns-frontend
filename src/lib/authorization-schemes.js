'use strict'

/**
 * This file contains the handlers for the authorization strategies used by hapi
 */

const Joi = require('joi')
const Client = require('../api/client')
const LicenceApi = require('../api/licence')

// Joi schema to validate a licence payload
const licenceSchema = Joi.object().keys({
  licence: Joi.string().alphanum().min(6).max(6).required(),
  postcode: Joi.string().required()
})

// Joi schema to validate an active directory login payload
const activeDirectorySchema = Joi.object().keys({
  user: Joi.string().email({ minDomainAtoms: 2 }).required(),
  password: Joi.string().required()
})

module.exports = {
  activeDirScheme: () => {
    return {
      options: {
        payload: true
      },

      payload: async (request, h) => {
        if (!request.payload.user || !request.payload.password) {
          return h.continue
        }

        request.payload.user = request.payload.user.replace(/\s+/g, '')

        const result = Joi.validate(request.payload, activeDirectorySchema,
          { allowUnknown: true, abortEarly: true })

        // If cannot validate the schema is not authorized
        if (result.error) {
          return h.continue
        }

        const auth = {
          username: request.payload.user,
          password: request.payload.password
        }

        try {
          await Client.request(auth, Client.method.GET, 'profile')
          request.app = {
            authorization: auth
          }
          return h.continue
        } catch (err) {
          if (Math.floor(Number.parseInt(err.statusCode) / 100) === 4) {
            return h.continue
          } else {
            throw err
          }
        }
      },

      /**
       * Hapi does not support authentication based on the payload as the authentication
       * system expected authentication system in the hearer and this is problematic where no javascript
       * is permitted. So here the authentication is passed on to the payload function which is intended
       * for schema validation
       * @param request
       * @param h
       * @returns {Promise<*>}
       */
      authenticate: async (request, h) => {
        return h.authenticated({ credentials: {} })
      }
    }
  },

  licenceScheme: () => {
    return {

      options: {
        payload: true
      },

      payload: async (request, h) => {
        if (!request.payload.licence || !request.payload.postcode) {
          return h.continue
        }

        request.payload.licence = request.payload.licence.replace(/\s+/g, '')
        request.payload.postcode = request.payload.postcode.replace(/\s+/g, '')

        const result = Joi.validate(request.payload, licenceSchema,
          { allowUnknown: true, abortEarly: true })

        // If cannot validate the schema is not authorized
        if (result.error) {
          return h.continue
        }

        const licence = request.payload.licence.toUpperCase()
        const postcode = request.payload.postcode.toUpperCase()

        try {
          const contact = await LicenceApi.getContactFromLicenceKey(request, licence, postcode)
          request.app = {
            authorization: { contactId: contact.contact.id }
          }
          return h.continue
        } catch (err) {
          if (Math.floor(Number.parseInt(err.statusCode) / 100) === 4) {
            return h.continue
          }
          throw err
        }
      },

      authenticate: async (request, h) => {
        return h.authenticated({ credentials: {} })
      }
    }
  }
}
