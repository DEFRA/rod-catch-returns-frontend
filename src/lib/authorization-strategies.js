'use strict'

/**
 * This file contains the handlers for the authorization strategies used by hapi
 */
module.exports = {
  sessionCookie: {
    cookie: {
      name: 'sid',
      password: process.env.COOKIE_PW,
      isSecure: process.env.HTTPS === 'true',
      clearInvalid: true
    },

    redirectTo: process.env.CONTEXT === 'ANGLER' ? '/licence-auth' : '/login',
    appendNext: process.env.CONTEXT === 'FMT',
    /**
     * validation function called on every request
     * When the cache-entry expires the user has to re-authenticate
     */
    validateFunc: async (request, session) => {
      const server = request.server
      const cached = await server.app.cache.get(session.sid)

      const out = {
        valid: !!cached
      }

      if (out.valid) {
        out.credentials = cached.authorization
      }

      return out
    }
  }
}
