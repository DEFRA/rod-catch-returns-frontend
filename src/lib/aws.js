'use strict'

/**
 * Handle the s3 bucket functions used for the FMT reports
 */
const { logger } = require('defra-logging-facade')
const AWS = require('aws-sdk')
const { S3 } = require('@aws-sdk/client-s3')
const Mime = require('./mime-desc')

// If the proxy details are set up then include them in the AWS configuration
logger.debug('checking for https proxy')
if (Object.keys(process.env).find(k => k === 'https_proxy')) {
  try {
    logger.debug(`Using proxy: ${process.env.https_proxy}`)
    const proxy = require('proxy-agent')
    AWS.config.update({
      httpOptions: {
        agent: proxy(process.env.https_proxy)
      }
    })
  } catch (err) {
    logger.error('Bad proxy specification: ' + err)
  }
}

logger.debug('initialising s3')
const s3 = new S3({ region: process.env.AWS_REGION || 'eu-west-1' })

// Convert the file name to a description
const fileNameToDesc = (filename) => {
  const desc = filename.replace(process.env.REPORTS_S3_LOCATION_FOLDER + '/', '').replace(/_/g, ' ').split('.')[0]
  return desc.charAt(0).toUpperCase() + desc.substr(1).toLowerCase()
}

const s3GetObjectTagging = async Key => {
  try {
    return await s3.getObjectTagging({
      Bucket: process.env.REPORTS_S3_LOCATION_BUCKET,
      Key
    })
  } catch (e) {
    logger.error(`Cannot retrieve report description: ${e}`)
    throw e
  }
}

/*
 * Get the report metadata for a given key. Use either a description tag
 * Or convert the filename to a description
 */
const getReportDescription = async key => {
  const data = await s3GetObjectTagging(key)
  const descTag = data.TagSet.find(k => k.Key.toLowerCase() === 'description')
  return {
    key,
    description: descTag ? descTag.Value.trim() : fileNameToDesc(key)
  }
}

const s3GetObject = async Key => {
  try {
    return await s3.getObject({
      Bucket: process.env.REPORTS_S3_LOCATION_BUCKET,
      Key
    })
  } catch (e) {
    logger.error(`Cannot retrieve report metadata: ${e}`)
    throw e
  }
}

/*
 * Get the report metadata for a given key. Use either a description tag
 * Or convert the filename to a description
 */
const getReportMetaData = async key => {
  const data = await s3GetObject(key)
  return {
    key,
    length: Math.round(data.ContentLength / 1000),
    lastModified: data.LastModified.toDateString(),
    contentType: Object.keys(Mime).includes(data.ContentType) ? Mime[data.ContentType] : data.ContentType
  }
}

const s3ListObjectsV2 = async () => {
  try {
    return await s3.listObjectsV2({
      Bucket: process.env.REPORTS_S3_LOCATION_BUCKET,
      Prefix: process.env.REPORTS_S3_LOCATION_FOLDER,
      StartAfter: `${process.env.REPORTS_S3_LOCATION_FOLDER}/`
    })
  } catch (e) {
    logger.error(`Cannot retrieve report listing ${e}`)
    throw e
  }
}

module.exports = {
  // Test that the specified S3 bucket exists
  reportLocationExists: async () => {
    try {
      logger.debug('report location exists', process.env.REPORTS_S3_LOCATION_BUCKET)
      const reportsLocationExtant = await s3.headBucket({ Bucket: process.env.REPORTS_S3_LOCATION_BUCKET })
      logger.debug('report location exists:', reportsLocationExtant)
      return reportsLocationExtant
    } catch (e) {
      logger.error(`Cannot find report location: ${e}`)
      throw e
    }
  },

  // List the available reports in the specified location.
  listReports: async () => {
    const data = await s3ListObjectsV2()
    const details = await Promise.all(data.Contents.map(c => getReportDescription(c.Key)))
    const reportMetaData = await Promise.all(details.map(d => getReportMetaData(d.key)))
    return details.map(d => Object.assign(d, reportMetaData.find(m => m.key === d.key)))
  },

  // Get a report by key
  getReport: async key => {
    try {
      return await s3.getObject({
        Bucket: process.env.REPORTS_S3_LOCATION_BUCKET,
        Key: key
      })
    } catch (e) {
      logger.error(`Cannot retrieve report: ${e}`)
      throw e
    }
  }
}
