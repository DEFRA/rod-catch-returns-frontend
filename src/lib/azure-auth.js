const msal = require('@azure/msal-node')

/** @type {msal.Configuration} */
const config = {
  auth: {
    clientId: process.env.MSAL_CLIENT_ID,
    clientSecret: process.env.MSAL_CLIENT_SECRET,
    authority: process.env.MSAL_ENDPOINT
  }
}

const msalClient = new msal.ConfidentialClientApplication(config)

const getAuthenticationUrl = async () => {
  const authUrl = await msalClient.getAuthCodeUrl({
    redirectUri: process.env.MSAL_REDIRECT_URI,
    responseMode: 'form_post'
  })
  return authUrl
}

module.exports = {
  getAuthenticationUrl
}
