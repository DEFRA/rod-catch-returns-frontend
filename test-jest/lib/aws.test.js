const mockS3 = Object.freeze({
  headBucket: jest.fn(),
  listObjectsV2: jest.fn(),
  getObject: jest.fn(),
  getObjectTagging: jest.fn()
})
process.env.https_proxy = 'https-proxy'
const { logger } = require('defra-logging-facade')
const aws = require('../../src/lib/aws')
const s3Client = require('@aws-sdk/client-s3')
const { NodeHttpHandler } = require('@aws-sdk/node-http-handler')
const Proxy = require('proxy-agent')

jest.mock('@aws-sdk/client-s3', () => ({
  S3: jest.fn(() => mockS3)
}))
jest.mock('defra-logging-facade')
jest.mock('proxy-agent')
jest.mock('@aws-sdk/node-http-handler', () => ({
  NodeHttpHandler: jest.fn()
}))

describe('aws', () => {
  describe('setup', () => {
    it('should pass region', async () => {
      expect(s3Client.S3).toHaveBeenCalledWith(
        expect.objectContaining({
          region: 'eu-west-1'
        })
      )
    })
    it('should setup proxy if https proxy defined in env vars', async () => {
      expect(s3Client.S3).toHaveBeenCalledWith(
        expect.objectContaining({
          requestHandler: expect.any(NodeHttpHandler)
        })
      )
    })
    it('should pass an instance of Proxy as the https agent', async () => {
      expect(NodeHttpHandler).toHaveBeenCalledTimes(1)
      expect(NodeHttpHandler).toHaveBeenCalledWith(
        expect.objectContaining({
          httpsAgent: expect.any(Proxy)
        })
      )
    })
    it('should only setup a single Proxy instance', async () => {
      expect(Proxy).toHaveBeenCalledTimes(1)
    })

    it('should use process.env.AWS_REGION env var if specified', async () => {
      jest.isolateModules(() => {
        jest.resetModules()
        const is3Client = require('@aws-sdk/client-s3')
        jest.mock('@aws-sdk/client-s3', () => ({
          S3: jest.fn(() => mockS3)
        }))
        process.env.AWS_REGION = 'any-old-where'

        require('../../src/lib/aws')

        expect(is3Client.S3).toHaveBeenCalledWith(
          expect.objectContaining({
            region: process.env.AWS_REGION
          })
        )
      })
    })

    it('should throw error if proxy spec is wrong', async () => {
      jest.isolateModules(() => {
        jest.resetModules()
        const is3Client = require('@aws-sdk/client-s3')

        jest.mock('@aws-sdk/client-s3', () => ({
          S3: jest.fn(() => mockS3)
        }))
        jest.mock('proxy-agent', () => jest.fn(() => { throw new Error('dodgy proxy') }))

        const loadAWS = () => require('../../src/lib/aws')
        expect(loadAWS).toThrow('Bad proxy specification: Error: dodgy proxy')
      })
    })

    it("shouldn't setup proxy if https proxy isn't defined in env vars", async () => {
      jest.isolateModules(() => {
        jest.resetModules()
        const is3Client = require('@aws-sdk/client-s3')

        jest.mock('@aws-sdk/client-s3', () => ({
          S3: jest.fn(() => mockS3)
        }))

        delete process.env.https_proxy
        const iaws = require('../../src/lib/aws')
        expect(is3Client.S3).not.toHaveBeenCalledWith(
          expect.objectContaining({
            requestHandler: expect.any(NodeHttpHandler)
          })
        )
      })
    })
  })

  describe('mock clearance tests', () => {
    beforeEach(() => {
      jest.clearAllMocks()
    })
    describe('reportLocationExists', () => {
      it('should resolve if the specified bucket exists', async () => {
        mockS3.headBucket.mockImplementationOnce(() => 'test')
        await expect(aws.reportLocationExists()).resolves.toBe('test')
      })

      it('should reject if the specified bucket does not exist', async () => {
        mockS3.headBucket.mockImplementationOnce(() => { throw new Error('error') })
        await expect(aws.reportLocationExists()).rejects.toThrow()
        expect(logger.error).toHaveBeenCalledTimes(1)
      })
    })

    describe('listReports', () => {
      it('should reject if it cannot retrieve the report listing', async () => {
        mockS3.listObjectsV2.mockImplementationOnce(async () => { throw new Error('error') })
        await expect(aws.listReports()).rejects.toThrow()
        expect(logger.error).toHaveBeenCalledTimes(1)
      })

      it('should resolve and return empty array if an empty array is returned from listObjectsV2', async () => {
        mockS3.listObjectsV2.mockImplementationOnce(async () => ({ Contents: [] }))
        await expect(aws.listReports()).resolves.toEqual([])
      })

      it('should return all objects provided by S3', async () => {
        mockS3.listObjectsV2.mockImplementationOnce(async () => ({ Contents: [{ Key: 'Report 1' }, { Key: 'Report 2' }] }))
        mockS3.getObjectTagging.mockImplementationOnce(async () => ({ TagSet: [{ Key: 'DeScRipTion', Value: 'Sample description for Report 1' }] }))
        mockS3.getObjectTagging.mockImplementationOnce(async () => ({ TagSet: [] }))
        mockS3.getObject.mockImplementationOnce(async () => ({ ContentLength: 20597, LastModified: new Date('2023-06-15T09:29:38.487Z'), ContentType: 'application/vnd.ms-excel' }))
        mockS3.getObject.mockImplementationOnce(async () => ({ ContentLength: 12345, LastModified: new Date('2023-06-15T09:58:08.102Z'), ContentType: 'Weird format spreadsheet' }))

        const reports = await aws.listReports()

        expect(reports).toMatchSnapshot()
      })

      it('reports errors thrown by getObject', async () => {
        mockS3.listObjectsV2.mockImplementationOnce(async () => ({ Contents: [{ Key: 'Report' }] }))
        mockS3.getObjectTagging.mockImplementationOnce(async () => ({ TagSet: [] }))
        mockS3.getObject.mockImplementationOnce(async () => { throw new Error('Tricky little timeout...') })

        const listReports = () => aws.listReports()

        await expect(listReports).rejects.toThrow('Cannot retrieve report metadata: Tricky little timeout...')
      })

      it('reports errors thrown by getObjectTagging', async () => {
        mockS3.listObjectsV2.mockImplementationOnce(async () => ({ Contents: [{ Key: 'Report' }] }))
        mockS3.getObjectTagging.mockImplementationOnce(async () => { throw new Error('Abandon ship!') })

        const listReports = () => aws.listReports()

        await expect(listReports).rejects.toThrow('Cannot retrieve report description: Abandon ship!')
      })
    })

    describe('getReport', () => {
      it('should return data from S3', async () => {
        const data = Symbol('data')
        mockS3.getObject.mockImplementationOnce(async () => data)
        await expect(aws.getReport()).resolves.toBe(data)
      })

      it('passes key to getObject', () => {
        const Key = Symbol('skeleton')
        aws.getReport(Key)
        expect(mockS3.getObject).toHaveBeenCalledWith(
          expect.objectContaining({ Key })
        )
      })

      it('uses REPORTS_S3_LOCATION_BUCKET env var as Bucket param', () => {
        const S3Bucket = 'S3 Location'
        process.env.REPORTS_S3_LOCATION_BUCKET = S3Bucket
        aws.getReport()
        expect(mockS3.getObject).toHaveBeenCalledWith(
          expect.objectContaining({ Bucket: S3Bucket })
        )
      })

      it('should throw error if getObject generates one', async () => {
        mockS3.getObject.mockImplementationOnce(async () => { throw new Error('error') })
        await expect(() => aws.getReport()).rejects.toThrow('error')
      })

      it('logs error if getObject generates one', async () => {
        mockS3.getObject.mockImplementationOnce(async () => { throw new Error('error') })
        try {
          await aws.getReport()
        } catch {}
        expect(logger.error).toHaveBeenCalledWith('Cannot retrieve report: Error: error')
      })
    })
  })
})
