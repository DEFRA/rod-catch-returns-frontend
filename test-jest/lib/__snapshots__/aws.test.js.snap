// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aws mock clearance tests listReports should return all objects provided by S3 1`] = `
Array [
  Object {
    "contentType": "Microsoft Excel",
    "description": "Sample description for Report 1",
    "key": "Report 1",
    "lastModified": "Thu Jun 15 2023",
    "length": 21,
  },
  Object {
    "contentType": "Weird format spreadsheet",
    "description": "Report 2",
    "key": "Report 2",
    "lastModified": "Thu Jun 15 2023",
    "length": 12,
  },
]
`;
